vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurityConfigurerAdapter.java

```diff
@@ -0,0 +1,105 @@
+package com.vaadin.flow.spring.security;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.server.HandlerHelper;
+import com.vaadin.flow.server.connect.EndpointUtil;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.util.matcher.OrRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+/**
+ * Provides basic Vaadin security configuration for the project.
+ * <p>
+ * Sets up security rules for a Vaadin application and restricts all URLs except
+ * for public resources and internal Vaadin URLs to authenticated user.
+ * <p>
+ * The default behavior can be altered by extending the public/protected methods
+ * in the class.
+ * <p>
+ * To use this, create your own web security configurer adapter class by
+ * extending this class instead of <code>WebSecurityConfigurerAdapter</code> and
+ * annotate it with <code>@EnableWebSecurity</code> and
+ * <code>@Configuration</code>.
+ * <p>
+ * For example <code>
+&#64;EnableWebSecurity
+&#64;Configuration
+public class MySecurityConfigurerAdapter extends VaadinWebSecurityConfigurerAdapter {
+    
+} 
+ * </code>
+ * 
+ */
+public abstract class VaadinWebSecurityConfigurerAdapter extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private VaadinDefaultRequestCache vaadinDefaultRequestCache;
+
+    @Autowired
+    private EndpointUtil endpointUtil;
+
+    @Autowired
+    private RequestUtil requestUtil;
```

Comment:
I don't think this direct dependency to fusion beans is okay in such a generic class. 

For example: We exclude fusion-endpoint from every application to reduce the huge amount of dependencies it depends on and we don't need as flow only user. Introducing the direct dependency here forces me to have fusion on the class path and even have a bean from fusion. 

Is this really necessary? 

Looking at the investigation in flow about separating flow and fusion, it doesn't really make sense to force spring user to have both of them again. 

CC: @pleku

Reply:
Wouldn't it be enough to mark this as `required=false`?

Sounds like the real problem is in the dependencies of Fusion though. There should be no need to manually optimize like that

Reply:
Using required false with a null check later on could work. 

Cleaning up the dependencies in fusion would be good too but I don't know if that's possible without a deep refactoring or improved production build. If I'm not mistaking we saved like 10-20mb just by excluding fusion from our final jar but I have to check again on my PC later.

Reply:
I got time to check it again:

- V19.0.3 (starter) with `fusion-endpoint` - `mvn clean package -Pproduction` took 43sec and the produced jar is ~54mb
- V19.0.3 (starter) **without** `fusion-endpoint` - `mvn clean package -Pproduction` took 21sec and the produced jar is ~39mb

Exclusion was done like this:

```xml
       <dependency>
            <groupId>com.vaadin</groupId>
            <!-- Replace artifactId with vaadin-core to use only free components -->
            <artifactId>vaadin</artifactId>
            <exclusions>
               <exclusion>
                 <groupId>com.vaadin</groupId>
                 <artifactId>fusion-endpoint</artifactId>
               </exclusion>
            </exclusions>
        </dependency>


```

Reply:
This is fixed in the PR where the code is actually introduced: #744

Reply:
@Artur- You are the best! Thank you :)